src/exponential_utils.jl

```diff
@@ -228,6 +238,9 @@ function arnoldi!(Ks::KrylovSubspace{B, T}, A, b::AbstractVector{T}; tol=1e-7,
   end
   V, H = getV(Ks), getH(Ks)
   vtol = tol * norm(A, Inf)
+  if iop == 0
+    iop = m
```

Comment:
in the PR you only show for 2? What does higher look like?

Reply:
For the output of `iop=p`, the Heisenberg matrix `H` will have a length-p upper band (i.e. only p non-zero diagonals for the upper part). I checked this before running the performance tests and it's working as intended.

Performance & accuracy wise, haven't really tested yet but the idea is by increasing `p` you get better accuracy and worse performance. For the almost symmetric operators that I showed in the tests, since IOP(2) is already good enough there's no need to check higher `p`s. For general operators I would expect a more visible performance-accuracy tradeoff for adjusting `p`.